server/src/plugins/queryFilter.js

```diff
@@ -0,0 +1,142 @@
+    method: ['GET'],
+    ignoreKeys: ['paging'],
+}
+
+// Holds Joi-schema descriptions, as these can be quite expensive to generate and this is a hot-path
+// cached by path as key
+
+// validateDescriptions is used to get queryParams with Filter-type
+const validateDescriptions = {}
+// renameDescriptions is used to rename the filter-Keys to the correct DB-column
+const renameDescriptions = {}
+
+const onPreHandler = function(request, h) {
+    const routeOptions = request.route.settings.plugins.queryFilter || {}
+
+    const options = {
+        ...this.options,
+        ...routeOptions,
+        ignoreKeys: this.options.ignoreKeys.concat(
+            routeOptions.ignoreKeys || []
+        ),
+    }
+
+    if (
+        !options.enabled ||
+        !options.method.includes(request.method.toUpperCase())
+    ) {
+        return h.continue
+    }
+
+    const routeQueryValidation = request.route.settings.validate.query
+    const rename = routeOptions.rename
+    const queryFilterKeys = new Set()
+    let renameMap = rename
+    let validateDescription = validateDescriptions[request.path]
+    let renameDescription = renameDescriptions[request.path]
+
+    if (rename) {
+        if (Joi.isSchema(rename)) {
+            if (!renameDescription) {
+                renameDescription = renameDescriptions[
+                    request.path
+                ] = rename.describe()
+            }
+            renameMap = renameDescription.renames.reduce((acc, curr) => {
+                const { from, to } = curr
+                acc[from] = to
+                return acc
+            }, {})
+        }
+    }
+
+    if (Joi.isSchema(routeQueryValidation)) {
+        if (!validateDescription) {
+            validateDescription = validateDescriptions[
+                request.path
+            ] = routeQueryValidation.describe()
+        }
+
+        // only add validations with .filter()
+        Object.keys(validateDescription.keys).forEach(k => {
+            const keyDesc = validateDescription.keys[k]
+            if (keyDesc.type === FILTER_TYPE) {
+                queryFilterKeys.add(k)
+            }
+        })
+    } else {
+        // add all keys if no validation
+        Object.keys(request.query).forEach(key => {
+            const val = request.query[key]
+            const parsed = parseFilterString(val)
+            request.query[key] = parsed
+            queryFilterKeys.add(key)
+        })
+    }
+
+    const queryFilters = Object.keys(request.query).reduce((acc, curr) => {
+        if (
+            queryFilterKeys.has(curr) &&
+            options.ignoreKeys.indexOf(curr) === -1
+        ) {
+            acc[curr] = request.query[curr]
+        }
+        return acc
+    }, {})
+
+    try {
+        const filters = new Filters(queryFilters, {
+            renameMap,
+        })
+        request.plugins.queryFilter = filters
+    } catch (e) {
+        Bounce.rethrow(e, 'system')
+        throw Boom.boomify(e, { statusCode: 400 })
```

Comment:
Wouldn't the Bounce.rethrow cause the Boom.boomify row to never run?

Reply:
No, Bounce just rethrows system errors, eg. syntaxErrors. It's so that we don't throw 400-errors for errors that really should be 500-errors.

Reply:
right, so when there is a system error this would throw in all other cases it would just skip and the Boom exception will be executed, I understand